internal/command/arguments/extended.go

```diff
@@ -155,11 +185,23 @@ func parseRawTargetsAndExcludes(targets []string, excludes []string) ([]addrs.Ta
 // Parse must be called on Operation after initial flag parse. This processes
 // the raw target flags into addrs.Targetable values, returning diagnostics if
 // invalid.
+//
+// Note to self: I wonder why this doesn't have a unit test. Investigate
+// the reason behind this, and potentially add one.
+// Many tests fail though, if this method is fucked up.
```

Comment:
I don't think there's a _great_ answer to this question. The argument parsing was, in much older versions of this codebase, duplicated into each separate command. It was refactored out here to this separate package later, but all of the tests for this functionality had originally lived in the per-command tests in `package command` and so I guess that's where they continue to live today.

I think it'd be nice to have some true unit tests for this `Operation` type in particular, and so I'd be happy if you wanted to add some as part of this work :grinning: but I think it's also fine to just continue with the way things are currently structured for now and do that refactoring in a separate PR, if you'd prefer to focus only on solving the problem that prompted this work.

Reply:
I'll see how long it takes me to deliver the actual feature, and then I'll evaluate then! But yes, I discovered what you referenced - that there most of the unit testing that would be implemented on this function already live in the child functions that this calls into.
Thanks for the background!